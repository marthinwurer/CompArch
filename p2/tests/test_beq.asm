;
; tests BEQ
;
; uses instructions MOV, BPT, CMP, ADD, BR, HALT
;
; uses addressing modes 0, 2, 6
;
	.globl	start
;
v1:	.word	0
v2:	.word	177777
v3:	.word	077777
;
start:	mov	v1,r1
	mov	v2,r2
	mov	v3,r3
	bpt
;
	cmp	r1,r1
	beq	ok1
	add	#1,r0
ok1:
;
	cmp	r1,r2
	beq	err1
	br	ok2
err1:	add	#2,r0
ok2:
;
	bpt
	halt
;
	.end	start
