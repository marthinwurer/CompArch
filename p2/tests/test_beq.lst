      1                                 ;
      2                                 ; tests BEQ
      3                                 ;
      4                                 ; uses instructions MOV, BPT, CMP, ADD, BR, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9                                 ;
     10 000000  000000                  v1:      .word   0
     11 000002  177777                  v2:      .word   177777
     12 000004  077777                  v3:      .word   077777
     13                                 ;
     14 000006  016701  177766          start:   mov     v1,r1
     15 000012  016702  177764                   mov     v2,r2
     16 000016  016703  177762                   mov     v3,r3
     17 000022  000003                           bpt
     18                                 ;
     19 000024  020101                           cmp     r1,r1
     20 000026  001402                           beq     ok1
     21 000030  062700  000001                   add     #1,r0
     22 000034                          ok1:
     23                                 ;
     24 000034  020102                           cmp     r1,r2
     25 000036  001401                           beq     err1
     26 000040  000402                           br      ok2
     27 000042  062700  000002          err1:    add     #2,r0
     28 000046                          ok2:
     29                                 ;
     30 000046  000003                           bpt
     31 000050  000000                           halt
     32                                 ;
     33 000052  000006'                          .end    start


err1     000042 R  
ok1      000034 R  
ok2      000046 R  
pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000006 RG 
v1       000000 R  
v2       000002 R  
v3       000004 R  
