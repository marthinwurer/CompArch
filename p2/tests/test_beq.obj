000 2 00 00
002 2 ff ff
004 2 ff 7f
006 4 c1 1d f6 ff
00a 4 c2 1d f4 ff
00e 4 c3 1d f2 ff
012 2 03 00
014 2 41 20
016 2 02 03
018 4 c0 65 01 00
01c 2 42 20
01e 2 01 03
020 2 02 01
022 4 c0 65 02 00
026 2 03 00
028 2 00 00
006
