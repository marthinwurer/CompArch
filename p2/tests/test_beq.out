CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 6
000006:  PS[00]  016701 177766  MOV 177766(R7),R1
000012:  PS[04]  016702 177764  MOV 177764(R7),R2
000016:  PS[10]  016703 177762  MOV 177762(R7),R3
000022:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000024]
000024:  PS[00]  020101  CMP R1,R1
000026:  PS[04]  001402  BEQ 4
000034:  PS[04]  020102  CMP R1,R2
000036:  PS[01]  001401  BEQ 2
000040:  PS[01]  000402  BR 4
000046:  PS[01]  000003  BPT
  R0[000000]  R1[000000]  R2[177777]  R3[077777]
  R4[000000]  R5[000000]  R6[000000]  R7[000050]
000050:  PS[01]  000000  HALT

Machine Halted - HALT instruction

Simulated time 78 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
